subprojects/launcher/src/main/java/org/gradle/launcher/daemon/server/exec/LogToClient.java

```diff
@@ -76,10 +79,18 @@ private AsynchronousLogDispatcher(DaemonConnection conn, final LogLevel buildLog
             this.connection = conn;
             this.listener = new OutputEventListener() {
                 public void onOutput(OutputEvent event) {
-                    if (dispatcher != null && event.getLogLevel() != null && event.getLogLevel().compareTo(buildLogLevel) >= 0) {
+                    if (dispatcher != null && (isMatchingBuildLogLevel(event) || isProgressEvent(event))) {
                         dispatcher.submit(event);
                     }
                 }
+
+                private boolean isProgressEvent(OutputEvent event) {
+                    return event instanceof ProgressStartEvent || event instanceof ProgressEvent || event instanceof ProgressCompleteEvent;
```

Comment:
we shouldn't check for specific types here. We should simply forward any event without a log level (as log level is not relevant) and any event with a log level that is >= the requested level.